app/api/adapters/db/flask_db.py

```diff
@@ -51,3 +55,61 @@ def health():
             db_client = flask_db.get_db(current_app)
     """
     return app.extensions[_FLASK_EXTENSION_KEY]
+
+
+Ts = TypeVarTuple("Ts")
+T = TypeVar("T")
+
+
+# https://github.com/python/mypy/issues/12280
+def with_db_session(f: Callable[[*Ts, db.Session], T]) -> Callable[[*Ts], T]:  # type: ignore
+    """Decorator for functions that need a database session.
+
+    This decorator will create a new session object and pass it to the function
+    as the first positional argument. The function's last parameter needs be named
+    "db_session". A transaction is not started automatically. If you want to start
```

Comment:
Should this say `The function's first parameter needs to be named`?

Reply:
oh yeah i need to clean this PR up for sure

btw i realize that mypy doesn't yet support variadic type variables, so one downside with the decorator is that the type checking won't be as strong

Reply:
oh nvm ignore the second part of that comment, i can use typing's ParamSpec and Concatenate